blog/2023-10-11-graphing-update.md

```diff
@@ -0,0 +1,301 @@
+---
+author: Alek Petuskey
+date: 2023-10-11
+title: "New Core Graphing Components"
+description: "Using Reflex's new core graphing feature to build a live streaming graphing app."
+image: /graphing.png
+---
+
+```python exec
+import reflex as rx
+import random
+from pcweb.base_state import State
+import asyncio
+# hacks because curly braces always evaluate python code
+pid = "{pid}"
+class self(object):
+    last_id = "{self.last_id}"
+```
+
+With the release of [Reflex v0.2.9](https://github.com/reflex-dev/reflex/releases/tag/v0.2.9) we've reworked the core graphing components from the ground up. The new components are more flexible and easier to use. 
+
+In this post we'll walk through the new components and show how to build a live streaming graphing app. This will also build on the [Unlocking New Workflows with Background Tasks](https://reflex.dev/blog/2023-09-28-unlocking-new-workflows-with-background-tasks/)  post, so if you haven't read that yet, check it out first.
+
+## Striking a Balance
+
+With our new graphing components we are aiming to strike a balance between flexibility and ease of use. We want to make it easy to build a graph, but we also want to make it easy to customize the graph to your needs.
+
+Check our the our [graphing docs](https://reflex.dev/docs/library/graphing/areachart/) for more details on the new components.
+
+
+## Livestreaming Example
+
+In this example we will create a live streaming graph that updates every second with random data.
+
+We start by defining some initial data for our chart to use as well as some imports we will use for the project:
+
+```python
+import reflex as rx
+import random
+import asyncio
+
+data = [
+    \{"name": "A", "uv": 10, "pv": 110, "amt": 210\},
+    \{"name": "B", "uv": 20, "pv": 120, "amt": 230\},
+    \{"name": "C", "uv": 30, "pv": 120, "amt": 240\},
+    \{"name": "D", "uv": 30, "pv": 130, "amt": 210\},
+    \{"name": "E", "uv": 20, "pv": 140, "amt": 230\},
+    \{"name": "F", "uv": 40, "pv": 170, "amt": 250\},
+    \{"name": "G", "uv": 50, "pv": 190, "amt": 260\},
+]
+```
+
+Next we define a `StreamingState` class that will be used to store the data and update it with an event handler:
+
+```python
+class StreamingState(State):
+    data = data
```

Comment:
Can we add some comments for these vars. And also maybe add type annotations to encourage users to use them